User-Level Threads/Create: cmake not found

The os-runner container does not contain `cmake` and `boost`. It can be manually installed using:
```
sudo apt update
sudo apt install cmake
sudo apt install libboost-all-dev
```

Moreover, after installing and running the commands from the lab, we get the following error:
```
 student@os:~/.../support/user-level-threads/build$ make
Scanning dependencies of target yield_barrier
[ 10%] Building CXX object CMakeFiles/yield_barrier.dir/yield_barrier.cc.o
[ 20%] Linking CXX executable yield_barrier
[ 20%] Built target yield_barrier
Scanning dependencies of target threads_and_fibers
[ 30%] Building CXX object CMakeFiles/threads_and_fibers.dir/threads_and_fibers.cc.o
[ 40%] Linking CXX executable threads_and_fibers
/usr/bin/ld: CMakeFiles/threads_and_fibers.dir/threads_and_fibers.cc.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/threads_and_fibers.dir/build.make:88: threads_and_fibers] Error 1
make[1]: *** [CMakeFiles/Makefile2:111: CMakeFiles/threads_and_fibers.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

This error can be solved by adding the following line to `CMakeLists.txt`:
```
SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

--- a/content/chapters/compute/lab/content/user-level-threads.md
+++ b/content/chapters/compute/lab/content/user-level-threads.md
@@ -8,6 +8,15 @@ We will use the fiber implementation from libboost.
 This implementation uses a cooperative scheduler on each thread, meaning that each fiber has to yield, in order for other fiber to be executed.
 We will also use C++, and the standard `thread` implementation.
 
+## Prerequisites
+
+Unless you are using the OS docker image, you will need to install `cmake` and `libboost`.
+You can do this with the following command:
+
+```console
+student@os:~$ sudo apt-get install cmake libboost-context-dev libboost-fiber-dev
+```
+
 ## Creation
 
 Follow the `support/user-level-threads/simple.cc` implementation.

--- a/content/chapters/compute/lab/support/user-level-threads/CMakeLists.txt
+++ b/content/chapters/compute/lab/support/user-level-threads/CMakeLists.txt
@@ -10,10 +10,12 @@ set(Boost_USE_MULTITHREADED      ON)
 set(Boost_USE_STATIC_RUNTIME    OFF)
 
 find_package(Boost REQUIRED COMPONENTS context fiber)
+find_package(Threads REQUIRED)
 
 include_directories(PUBLIC ${Boost_INCLUDE_DIRS})
 link_libraries(Boost::context)
 link_libraries(Boost::fiber)
+link_libraries(Threads::Threads)
 add_compile_definitions(PUBLIC BOOST_STACKTRACE_USE_ADDR2LINE)
 
 add_executable(simple simple.cc)

--- a/util/os-runner/Dockerfile
+++ b/util/os-runner/Dockerfile
@@ -42,7 +42,10 @@ RUN apt-get update &&   \
     bind9-host  \
     net-tools   \
     netcat  \
-    libssl-dev
+    libssl-dev \
+    cmake \
+    libboost-context-dev \
+    libboost-fiber-dev
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 

